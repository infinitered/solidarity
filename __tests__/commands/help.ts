import helpCommand from '../../src/commands/help'

const context = {
  print: {
    info: jest.fn(),
    printCommands: jest.fn(),
    success: jest.fn(),
    colors: {
      magenta: jest.fn()
    }
  }
}

test('Snapshot check help command', () => {
  expect(helpCommand).toMatchSnapshot()
})

test('Enforce required properties', () => {
  expect(helpCommand.description).toBeTruthy()
  expect(helpCommand.run).toBeTruthy()
  expect(typeof helpCommand.run).toBe('function')
})

test('Calls print items several times', () => {
  expect(context.print.info.mock.calls.length).toBe(0)
  expect(context.print.printCommands.mock.calls.length).toBe(0)
  expect(context.print.success.mock.calls.length).toBe(0)
  expect(context.print.colors.magenta.mock.calls.length).toBe(0)
  helpCommand.run(context)
  expect(context.print.info.mock.calls.length).toBe(6)
  expect(context.print.printCommands.mock.calls.length).toBe(1)
  expect(context.print.success.mock.calls.length).toBe(2)
  expect(context.print.colors.magenta.mock.calls.length).toBe(2)
})
