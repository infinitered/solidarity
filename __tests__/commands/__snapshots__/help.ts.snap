// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot check help command 1`] = `
Object {
  "alias": "h",
  "description": "Displays this help",
  "run": [Function],
}
`;
